dyno-queues-redis/src/main/java/com/netflix/dyno/queues/shard/ConsistentDynoShardSupplier.java

```diff
@@ -0,0 +1,54 @@
+package com.netflix.dyno.queues.shard;
+
+import com.netflix.dyno.connectionpool.Host;
+import com.netflix.dyno.connectionpool.HostSupplier;
+import com.netflix.dyno.queues.ShardSupplier;
+
+import java.util.*;
+
+abstract class ConsistentDynoShardSupplier implements ShardSupplier {
+
+    protected HostSupplier hs;
+
+    protected  String region;
+
+    protected String localRack;
+
+    protected Map<String, String> rackToShardMap;
+
+    /**
+     * Dynomite based shard supplier.  Keeps the number of shards in parity with the hosts and regions
+     * @param hs Host supplier
+     * @param region current region
+     * @param localRack local rack identifier
+     */
+    public ConsistentDynoShardSupplier(HostSupplier hs, String region, String localRack) {
+        this.hs = hs;
+        this.region = region;
+        this.localRack = localRack;
+    }
+
+    public void setRackToShardMap(Map<String, String> rackToShardMapEntries) {
+        rackToShardMap = new HashMap<>(rackToShardMapEntries);
+    }
+
+    @Override
+    public String getCurrentShard() {
+        return rackToShardMap.get(localRack);
+    }
+
+    @Override
+    public Set<String> getQueueShards() {
+        Set<String> queueShards = new HashSet<>();
+        List<Host> hosts = hs.getHosts();
+        for (Host host : hosts) {
```

Comment:
need not be changed. but, can be simplified to `return hs.getHosts().stream().map(host -> rackToshardMap.get(host.getRack()).collect(Collectors.toSet());`

Reply:
Thanks for the suggestion. Turns out that this has some syntax errors. Will forego for now and update later.